### PR TITLE
Added a layer for talking to BigQuery with tests.

### PR DESCRIPTION
**Pull request by fernandalavalle**
This change is [Reviewable](https://reviewable.io/reviews/m-lab/signal-searcher/8)

### CODE REVIEW

**Comment by pboothe** [comment]
Please add httplib2 to the requirements.txt file, so that Travis can run the checks!

File: bigquery_talk.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                    time.sleep(5)
+                elif status == 'PENDING':
+                    logger.info(
+                        'Waiting for query to submit, spent %d seconds so '
+                        'far.', time_waiting)
+                    time.sleep(10)
+                elif status == 'DONE':
+                    logger.info('Query status complete.')
+                    return self._collect_query_results()
+                    break
```
Comment (pboothe) [inline bigquery_talk.py]: break after return is redundant.

File: bigquery_talk.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                else:
+                    raise Exception('UnknownBigQueryResponse')
+        return None
+
+    def _collect_query_results(self):
+        try:
+            response = self._job.getQueryResults(projectId=self._project_id, jobId=self._job_id).execute()
+            response_tuple = (str(response['schema']['fields'][0]['name']), int(response['rows'][0]['f'][0]['v']))
```
Comment (pboothe) [inline bigquery_talk.py]: ?

Can we write our queries better so that this line makes more sense?

**Comment by pboothe** [comment]
Also, a request to increase the coverage. I would happily meet over VC and chat about how to run the code-coverage tool locally so that you can assess your code coverage without having to go through Travis CI

**Comment by fernandalavalle** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[bigquery_talk.py, line 225 at r2](https://reviewable.io:443/reviews/m-lab/signal-searcher/8#-KRnmYqQISugLIEUcPuA-r2-225:-KSH_Kb5VBcWJ0GffYOs:b-u5v58c) ([raw file](https://github.com/m-lab/signal-searcher/blob/3797dc430ac724701234012cd3d9a6c7eb60a454/bigquery_talk.py#L225)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary>
> ?
> 
> 
> 
> Can we write our queries better so that this line makes more sense?
</details>
sure. what do you mean by that?  how do I tweak the response format?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/signal-searcher/8)*

**Comment by pboothe** [comment]
*[bigquery_talk.py, line 225 at r2](https://reviewable.io:443/reviews/m-lab/signal-searcher/8#-KRnmYqQISugLIEUcPuA-r2-225:-KSIBRJK5XWGYCbNMi0f:blti9b2) ([raw file](https://github.com/m-lab/signal-searcher/blob/3797dc430ac724701234012cd3d9a6c7eb60a454/bigquery_talk.py#L225)):*
<details><summary><i>Previously, fernandalavalle wrote…</i></summary>
> sure. what do you mean by that?  how do I tweak the response format?
</details>
Can you give things better names so that we aren't looking for 'f' and 'v'?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/signal-searcher/8)*

**Comment by pboothe** [comment]
*[bigquery_talk.py, line 225 at r2](https://reviewable.io:443/reviews/m-lab/signal-searcher/8#-KRnmYqQISugLIEUcPuA-r2-225:-KSICXLcGFzp2lb21yD3:b7o4b81) ([raw file](https://github.com/m-lab/signal-searcher/blob/3797dc430ac724701234012cd3d9a6c7eb60a454/bigquery_talk.py#L225)):*
<details><summary><i>Previously, pboothe (Peter Boothe) wrote…</i></summary>
> Can you give things better names so that we aren't looking for 'f' and 'v'?
</details>
If you don't have any unnamed function results in your SELECT statements, please ignore this comment. I definitely don't want you to have to do thing in any way other than the way BigQuery naturally works, so if those key names are populated internally, then don't worry about them.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/signal-searcher/8)*

**Comment by pboothe** [comment]
LGTM. Can the next CL please be an integration with this code and `signal-searcher.py` or `mlabreader.py` ?

**Pull request merged**